Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM golang:1.21 AS builder
+
+WORKDIR /building
+COPY . .
+
+ARG APP
+RUN make ${APP}
+
+FROM alpine:3.18 AS runtime
+
+ARG APP
+RUN addgroup -g 1000 -S ${APP} && adduser -u 1000 -S ${APP} -G ${APP} --home /app \
```

Comment:
Given that frp is often used to listen on port 80, will not using the root user have any impact? On the other hand, what problems may arise from using the root user?

Reply:
Using root opens the attack surface to a whole lot more security bug possibilities.
The privileged port problem was solved 3 years ago:
https://stackoverflow.com/questions/66431299/i-can-bind-to-port-80-as-a-non-root-user-in-a-docker-container-why-whats-goin 
Can you try both new images to verify the expected functionality?